src/domUpdates.js

```diff
@@ -62,10 +65,108 @@ const renderGrid = () => {
   recipeGrid.innerHTML = createGridHTML(gridData);
 }
 
-// Event listeners
-window.addEventListener("load", renderGrid)
+const getTagsFromRecipes = recipes => {
+  const uniqueTags = [];
+  const allTags = recipes.flatMap(recipe => recipe.tags)
+  allTags.forEach(tag => {
+    if (!uniqueTags.includes(tag)) {
+      uniqueTags.push(tag);
+    }
+  })
+  return uniqueTags;
+}
+
+const clubTagsAndIcons = tags => {
+  const tagsAndIcons = tags.map((tag, index) => {
+    return {
+      name: tag,
+      path: `./images/${tag}.png`,
+      row: (index + 1) % 2
+    };
+  });
+
+  return tagsAndIcons;
+}
+
+const splitTagsinRows = tagsAndIcons => {
+  const topRow = tagsAndIcons.filter(tag => tag.row === 1);
+  const bottomRow = tagsAndIcons.filter(tag => tag.row === 0);
+  return [topRow, bottomRow];
+}
+
+const createTagCardHTML = tag => {
+  let htmlCode = '';
+  htmlCode += `
+  <div class = "tag-card">
+      <div class="tag-image-bg">
+          <img class = "tag-image" src = "${tag.path}">
+      </div>
+      <div class="tag-option">
+          <p class="tag-text">${tag.name}</p>
```

Comment:
same comment from earlier. If we decide to remove the surrounding `tag-option` div we will need to make that change here as well

Reply:
Yes :) done